lib/truffle/socket/basic_socket.rb

```diff
@@ -166,7 +165,9 @@ def send(message, flags, dest_sockaddr = nil)
         end
       end
 
-      return buf.read_string(n_bytes)
+      str = buf.read_string(n_bytes)
+      buffer.replace(str) if buffer
+      str
```

Comment:
The correct pattern here seems to be:
```ruby
buffer ? buffer.replace(str) : str
```
because the buffer should be returned, not `str`, if a buffer is given.

Reply:
done